opslevel/resource_opslevel_integration_google_cloud.go

```diff
@@ -0,0 +1,277 @@
+		InstalledAt:           ComputedStringValue(googleCloudIntegration.InstalledAt.UTC().Format(time.RFC3339)),
+		Name:                  RequiredStringValue(googleCloudIntegration.Name),
+		OwnershipTagKeys:      StringSliceToSetValue(googleCloudIntegration.GoogleCloudIntegrationFragment.OwnershipTagKeys),
+		PrivateKey:            givenModel.PrivateKey,
+		TagsOverrideOwnership: types.BoolValue(googleCloudIntegration.GoogleCloudIntegrationFragment.TagsOverrideOwnership),
+	}
+
+	projects := make([]googleCloudProjectResourceModel, len(googleCloudIntegration.Projects))
+	for i, project := range googleCloudIntegration.Projects {
+		projects[i] = googleCloudProjectResourceModel{
+			ID:   RequiredStringValue(project.ID),
+			Name: RequiredStringValue(project.Name),
+			URL:  RequiredStringValue(project.URL),
+		}
+	}
+	projectsList, tmp := types.ListValueFrom(ctx, types.ObjectType{AttrTypes: googleCloudProjectAttrs()}, projects)
+	diags.Append(tmp...)
+	if diags.HasError() {
+		return integrationGoogleCloudResourceModel{}
+	}
+	resourceModel.Projects = projectsList
+
+	return resourceModel
+}
+
+func (r *integrationGoogleCloudResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_integration_google_cloud"
+}
+
+func (r *integrationGoogleCloudResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Google Cloud Integration resource",
+
+		Attributes: map[string]schema.Attribute{
+			"aliases": schema.ListAttribute{
+				Description: "All of the aliases attached to the integration.",
+				Computed:    true,
+				ElementType: types.StringType,
+			},
+			"client_email": schema.StringAttribute{
+				Description: "The service account email OpsLevel uses to access the Google Cloud account.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"private_key": schema.StringAttribute{
+				Description: "The private key for the service account that OpsLevel uses to access the Google Cloud account.",
+				Required:    true,
+				Sensitive:   true,
+			},
+			"created_at": schema.StringAttribute{
+				Description: "The time this integration was created.",
+				Computed:    true,
+			},
+			"id": schema.StringAttribute{
+				Description: "The ID of the integration.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"installed_at": schema.StringAttribute{
+				Description: "The time that this integration was successfully installed, if null, this indicates the integration was not completely installed.",
+				Computed:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the integration.",
+				Required:    true,
+			},
+			"ownership_tag_keys": schema.SetAttribute{
```

Comment:
What is the rational for using a set here?  Does the API actually treat it like a set under the hood? (as in if i send to the API `["one", "two" ,"three", "two"]` it will return `["one", "two" ,"three"]`)

Reply:
I think that this is just carried over from us handling lists of aliases as sets - can probably change this

Reply:
<img width="1176" alt="image" src="https://github.com/user-attachments/assets/eaf9c671-21b9-4879-9117-97512204119e">

Can confirm that the ordering provided by the user is preserved by the API and so there is no reason to use a set here.

Reply:
but you didn't test duplicates - which is what a set is about - eliminating duplicates

Reply:
@rocktavious in the case where there are duplicates, the API call will fail.
<img width="1182" alt="image" src="https://github.com/user-attachments/assets/e2b54756-1657-422c-8550-8a1f94eb6294">

I thought to initially use a set to prevent ordering differences between the API and what the user's input is to prevent the loops we've seen in the past with other array fields where the users' configs had `["foo", "bar"]` but the API would return `["bar", "foo"]` causing Terraform to complain about mismatch in the state

Reply:
@rocktavious sets have been removed completely from this integration resource's code, schema and tests

Reply:
Does terraform have a validation function on lists for duplicate values?  This way at plan time we can raise this validation error the API is going to if they have duplicates instead of having to wait for apply - helping the customer out so that terraform plan can sort of predict what the API is going to do and give them faster iteration.

Sounds like we might need to have a knowledge transfer on basic collection types to ensure your mental model of them is correct. A `set` is a uniqueness constraint on the list of items.  A `set` has nothing to do with the ordering of the values in the underlying list of items.  Usually that is implemented as a different type called an `OrderedSet` which is not very common but some languages have it.

Reply:
Checked again there is `listvalidtor.UniqueValues()` that I have added here